docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,279 @@
+
+#### Schema
+
+```
+    CREATE TABLE system.job_status_progress (
+    	job_id INT NOT NULL,
+    last_updated TIMESTAMPTZ,
+     	percent_complete FLOAT, # null if job does not have a defined completion percentage
+    	timestamp_complete DECIMAL, # set only for jobs that advance in time e.g. cdc or replication
+    	PRIMARY KEY (job_id, last_updated DESC)
+    )
+    CREATE TABLE system.job_status_message (
+    	job_id INT NOT NULL,
+    last_updated TIMESTAMPTZ,
+    	status STRING, # like v1's running_status, e.g. "waiting for leases" or "importing rows"
+    	detail STRING, # more info, reason, etc.
+    	PRIMARY KEY (job_id, last_updated DESC)
+    )
+```
+
+
+
+#### SHOW JOBS
+
+The job control table, along with these tables plus the intake table, defines the set of jobs the system is running, will run or has has failed or creased running, along with human-readable information about them, such as their description, percentage complete, or  recent status messages; taken together, these what an operator wants to know when they ask "what jobs is my cluster running? How complete are the job I was running? Did they fail and if so why?"  This interaction is fulfilled by SHOW JOBS and the jobs page on the DB console. Both of these are powered by an internal virtual table crdb_internal.jobs, and will continue to be going forward, however this table will now be reading columns only from the control table and the status tables (plus the intake table). Rendering the virtual table will _not_ require decoding large proto payloads, that also contain large amounts of jon-specific state, nor will it block on transactions that are manipulating that state, as the virtual table will not depend on the info table, described below, where that state now resides and is manipulated.
+
+In contrast, job-specific detail inspection, such as SHOW BACKUP JOB or similar code called from the per-job DB console page, will delegate to job-specific code that can in turn determine which keys from the info table it should fetch and render, to show more detailed, job-type-specific information. However for the common inspection tools of SHOW JOBS and the DB Console that show an overview of many different types of jobs at once will exclusively be iterating over these tables where non-control transactions are forbidden and all fields are native SQL columns, not large nested protocol buffer messages, reduce the risk of SHOW JOBSs hanging or consuming excessive memory.
+
+SHOW JOBS will additionally need to union rows in the intake table described below, using SKIP LOCKED, as they are committed runnable jobs that just have not yet been moved into the control table.
+
+
+#### Alternatives considered
+
+
+###### Status and progress columns in a single row that are updated in place
```

Comment:
This nicely explains why the status and progress columns are not updated in place, but I don't understand why there isn't a single `job_status` table. Will it be common to want either just the `progress` or just the human readable status? The note elsewhere about the progress vs message status data being different sizes didn't resonate for me. Why is that a problem? Even with the separate `job_status_message` table, the rows in that table might be vastly different sizes due to the messages being different sizes. If that isn't a problem in the `job_status_message` table, why is a problem in a unified `job_status` table?